modules/host/storagefolders_smoke_test.go

```diff
@@ -0,0 +1,877 @@
+		}
+
+		// For every sector in the sector usage map, make sure the database has
+		// a matching sector with the right expiry information.
+		for sectorRoot, expiryHeights := range sectorUsageMap {
+			usageBytes := bsu.Get(ht.host.sectorID(sectorRoot[:]))
+			if usageBytes == nil {
+				return errors.New("no usage info on known sector")
+			}
+			var usage sectorUsage
+			err := json.Unmarshal(usageBytes, &usage)
+			if err != nil {
+				return err
+			}
+			if len(usage.Expiry) != len(expiryHeights) {
+				return errors.New("usage information mismatch")
+			}
+			for i, expiryHeight := range usage.Expiry {
+				if expiryHeight != expiryHeights[i] {
+					// The correctness could be made not-implementation
+					// dependent by sorting the two arrays before comparing
+					// them, but that was deemed an unneeded step for this
+					// test.
+					return errors.New("usage expiry height mismatch - correctness is implementation dependent")
+				}
+			}
+		}
+		return nil
+	})
+}
+
+// TestStorageFolderUsage is a general integration test which tries all of the
+// major storage folder operations in various orders, all while adding and
+// removing sectors to verify that the behavior works as expected.
+func TestStorageFolderUsage(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	t.Parallel()
+	ht, err := newHostTester("TestStorageFolderUsage")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Start by checking that the initial state of the host has no storage
+	// added to it.
+	totalStorage, remainingStorage, err := ht.host.Capacity()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if totalStorage != 0 || remainingStorage != 0 {
+		t.Error("initial capacity of host is not reported at 0 - but no drives have been added!")
+	}
+
+	// Try adding a sector when there are no storage folders.
+	// Host needs to be locked when the unexported sector function is being
+	// used.
+	sectorRoot, sectorData, err := createSector()
+	if err != nil {
+		t.Fatal(err)
+	}
+	ht.host.mu.Lock()
+	err = ht.host.addSector(sectorRoot, 10, sectorData)
+	ht.host.mu.Unlock()
+	if err != errNoStorage {
+		t.Fatal(err)
+	}
+
+	// Add a storage folder, simulating a new drive being connected to the
+	// host.
+	storageFolderOne := filepath.Join(ht.persistDir, "host drive 1")
+	// Try using a file size that is too small. Because a filesize check is
+	// quicker than a disk check, the filesize check should come first.
+	err = ht.host.AddStorageFolder(storageFolderOne, minimumStorageFolderSize-1)
+	if err != errSmallStorageFolder {
+		t.Fatal("expecting errSmallStorageFolder:", err)
+	}
+	// Try linking to a storage folder that does not exist.
+	err = ht.host.AddStorageFolder(storageFolderOne, minimumStorageFolderSize)
+	if err == nil {
+		t.Fatal("should not be able to link to a storage folder which does not exist")
+	}
+	// Try linking to a storage folder that's not a directory.
+	err = ioutil.WriteFile(storageFolderOne, make([]byte, minimumStorageFolderSize), 0700)
```

Comment:
I noticed that `ioutil` is being used in a lot of places instead of `dependencies`